drivers/dma/dma_stm32u5.c

```diff
@@ -349,10 +351,20 @@ static int dma_stm32_configure(const struct device *dev,
 				&dev_config->streams[id - STM32_DMA_STREAM_OFFSET];
 	DMA_TypeDef *dma = (DMA_TypeDef *)dev_config->base;
 	LL_DMA_InitTypeDef DMA_InitStruct;
-	int ret;
-
 	LL_DMA_StructInit(&DMA_InitStruct);
 
+#if DT_HAS_COMPAT_STATUS_OKAY(st_stm32u5_dma)
+	/*  Linked list Node  and structure initialization */
+	LL_DMA_InitNodeTypeDef NodeConfig;
+	static LL_DMA_LinkNodeTypeDef Node_GPDMA_Channel;
+	LL_DMA_InitLinkedListTypeDef DMA_InitLinkedListStruct;
```

Comment:
Declarations should be done at the begining of a code block (before `LL_DMA_StructInit(&DMA_InitStruct);` in this case)